client/app/services/query-result.js

```diff
@@ -103,6 +103,10 @@ function QueryResultService($resource, $timeout, $q) {
             } else if (isString(v) && v.match(/^\d{4}-\d{2}-\d{2}$/)) {
               row[k] = moment.utc(v);
               newType = 'date';
+            } else if (v.match(/^\d{4}\d{2}\d{2}$/)) {
+              const newlyFormatedDate = v.slice(0, 4) + '-' + v.slice(4, 6) + '-' + v.slice(6);
```

Comment:
Instead of constructing a new string, you can also do `moment.utc(v, "YYYYMMDD")`.